app/models/lock.rb

```diff
@@ -120,4 +121,8 @@ def nil_out_blank_resource_type
   def unique_global_lock
     errors.add(:resource_id, :invalid) if global? && Lock.global.first
   end
+
+  def valid_delete_at
+    errors.add(:delete_at, 'Date must be in the future') unless delete_at.nil? || delete_at.future?
```

Comment:
prefer if for multiple conditions `if !delete_at&.future?`

Reply:
Rubocop seems to prefer the `unless` syntax here

Reply:
how about `if delete_at && !delete_at.future?`

Reply:
how about `if delete_at&.past?` 🎉